Issue with kanidm_lib_file_permissions when building on FreeBSD

### I did this

```
root@idm:~/kanidm # make release/kanidmd
cargo build -p daemon --bin kanidmd --release
   Compiling libsqlite3-sys v0.25.2
   Compiling kanidmd_lib v1.2.3 (/root/kanidm/server/lib)
   Compiling rustix v0.38.34
   Compiling kanidm_lib_file_permissions v1.2.3 (/root/kanidm/libs/file_permissions)
   Compiling cookie_store v0.20.0
   Compiling kanidmd_core v1.2.3 (/root/kanidm/server/core)
   Compiling hyper-tls v0.5.0
   Compiling axum-server v0.5.1
   Compiling daemon v1.2.3 (/root/kanidm/server/daemon)
   Compiling utoipa-swagger-ui v4.0.0
error[E0599]: no method named `st_gid` found for reference `&Metadata` in the current scope
  --> libs/file_permissions/src/unix.rs:22:22
   |
22 |     let f_gid = meta.st_gid();
   |                      ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_gid` is implemented but not in scope; perhaps you want to import it
   |
1  + use std::os::freebsd::fs::MetadataExt;
   |
help: there is a method `st_uid` with a similar name
   |
22 |     let f_gid = meta.st_uid();
   |                      ~~~~~~

error[E0599]: no method named `st_uid` found for reference `&Metadata` in the current scope
  --> libs/file_permissions/src/unix.rs:23:22
   |
23 |     let f_uid = meta.st_uid();
   |                      ^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_uid` is implemented but not in scope; perhaps you want to import it
   |
1  + use std::os::freebsd::fs::MetadataExt;
   |
help: there is a method `st_gid` with a similar name
   |
23 |     let f_uid = meta.st_gid();
   |                      ~~~~~~

error[E0599]: no method named `st_mode` found for reference `&Metadata` in the current scope
  --> libs/file_permissions/src/unix.rs:25:23
   |
25 |     let f_mode = meta.st_mode();
   |                       ^^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_mode` is implemented but not in scope; perhaps you want to import it
   |
1  + use std::os::freebsd::fs::MetadataExt;
   |
help: there is a method `mode` with a similar name
   |
25 |     let f_mode = meta.mode();
   |                       ~~~~

error[E0599]: no method named `st_gid` found for struct `Metadata` in the current scope
   --> libs/file_permissions/src/unix.rs:176:34
    |
176 |                 let f_gid = meta.st_gid();
    |                                  ^^^^^^
    |
   ::: /usr/local/lib/rustlib/src/rust/library/std/src/os/freebsd/fs.rs:40:8
    |
40  |     fn st_gid(&self) -> u32;
    |        ------ the method is available for `Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_gid` is implemented but not in scope; perhaps you want to import it
    |
1   + use std::os::freebsd::fs::MetadataExt;
    |
help: there is a method `st_uid` with a similar name
    |
176 |                 let f_gid = meta.st_uid();
    |                                  ~~~~~~

error[E0599]: no method named `st_uid` found for struct `Metadata` in the current scope
   --> libs/file_permissions/src/unix.rs:177:34
    |
177 |                 let f_uid = meta.st_uid();
    |                                  ^^^^^^
    |
   ::: /usr/local/lib/rustlib/src/rust/library/std/src/os/freebsd/fs.rs:38:8
    |
38  |     fn st_uid(&self) -> u32;
    |        ------ the method is available for `Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_uid` is implemented but not in scope; perhaps you want to import it
    |
1   + use std::os::freebsd::fs::MetadataExt;
    |
help: there is a method `st_gid` with a similar name
    |
177 |                 let f_uid = meta.st_gid();
    |                                  ~~~~~~

error[E0599]: no method named `st_mode` found for struct `Metadata` in the current scope
   --> libs/file_permissions/src/unix.rs:178:35
    |
178 |                 let f_mode = meta.st_mode();
    |                                   ^^^^^^^
    |
   ::: /usr/local/lib/rustlib/src/rust/library/std/src/os/freebsd/fs.rs:34:8
    |
34  |     fn st_mode(&self) -> u32;
    |        ------- the method is available for `Metadata` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `MetadataExt` which provides `st_mode` is implemented but not in scope; perhaps you want to import it
    |
1   + use std::os::freebsd::fs::MetadataExt;
    |
help: there is a method `mode` with a similar name
    |
178 |                 let f_mode = meta.mode();
    |                                   ~~~~

For more information about this error, try `rustc --explain E0599`.
error: could not compile `kanidm_lib_file_permissions` (lib) due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
*** Error code 101

Stop.
make: stopped in /root/kanidm
```

### I expected the following

Kanidm to build properly

### This happened instead
```
error: could not compile `kanidm_lib_file_permissions` (lib) due to 6 previous errors 
```

### Kanidm version details
```
git clone -b v1.2.3 https://github.com/kanidm/kanidm.git
```


- Output of `kanidm(d) version`: None yet, as I'm still building
- Are you running it in a container? If so, which image/tag?: Kind of :) see next question.
- If not a container, how'd you install it: FreeBSD Jail
- Operating System / Version (On Unix please post the output of `uname -a`): `FreeBSD idm.bsd.am 14.0-RELEASE-p6 FreeBSD 14.0-RELEASE-p6 #0: Tue Mar 26 20:26:20 UTC 2024     root@amd64-builder.daemonology.net:/usr/obj/usr/src/amd64.amd64/sys/GENERIC amd64`

### Any other comments

Just thank you for this awesome project!

--- a/libs/file_permissions/src/unix.rs
+++ b/libs/file_permissions/src/unix.rs
@@ -1,5 +1,8 @@
 use std::fs::Metadata;
 
+#[cfg(target_os = "freebsd")]
+use std::os::freebsd::fs::MetadataExt;
+
 #[cfg(target_os = "linux")]
 use std::os::linux::fs::MetadataExt;
 

